### ISSUE TITLE
Integrate with Discord RPC

### ISSUE THREAD

**Issue by ghost** [DiNitride/Discord-Music-Status#50]
Now that Discord supports new features with Rich Presence, can this be updated to work with it? I use iTunes for music but would still like to share what I'm listening to.

**Comment by DiNitride** [comment]
I'd love to, I messed around with it once actually. Unfortunately, there's an issue - To use RPC, you have to use an application registered with Discord, the same way bots are. Discord actually get's the "playing" name from the title of the application. So in this case, it would be "Playing Discord-Music-Status" which, doesn't look very good.

To have individual "Playing iTunes" or "Playing Winamp" I would have to have a different application for each, which isn't viable.

One potential solution would be to run a single app just called "Music" or something and have it say "Playing Music" (or "Listening to" if I can do that in any way with RPC), and have that specify the song and player in the details.

Thoughts?

Also side note this does support iTunes already, Snip can read from iTunes.

**Comment by RobertPaulson90** [comment]
Could it potentially be possible to mimic the Spotify API too?
![image](https://user-images.githubusercontent.com/1857719/36990790-04ae8a1c-20a6-11e8-852e-cc7558e84ef0.png)

**Comment by ghost** [comment]
> One potential solution would be to run a single app just called "Music" or something and have it say "Playing Music" (or "Listening to" if I can do that in any way with RPC), and have that specify the song and player in the details.

I think that would be a good idea. I don't *think* you can get "Listening to" through the RPC but just playing "Music" or something similar would be fine. It could basically be the same as Spotify's integration (showing progress through the song, artist, song name etc.) just without the listen along feature.

> Could it potentially be possible to mimic the Spotify API too?

I think that is a specific integration between Spotify and Discord which wouldn't be able to be accessed by a regular dev.

**Comment by DiNitride** [comment]
There is a way to get "Listening to", 90% sure. You can do it normally so I should be able to do it with RPC.

> Could it potentially be possible to mimic the Spotify API too?

Above comment got right basically, I can mimic it in some similar ways, but not all.

**Comment by ghost** [comment]
You’re right. I found [this](https://www.reddit.com/r/discordapp/comments/7q9ndr/comment/dsog7sa) which may be useful to you?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by DiNitride** [comment]
Thanks, I've actually already done with with the API for bots, this program
does say "Listening to". However I'm unsure if it works differently when
using RPC, and whether it is available via RPC yet
